proposals/infra/INF-0004-register-types-and-diagnostics.md

```diff
@@ -55,7 +55,7 @@ names and diagnostic messages that are relevant in this spec:
 | err_hlsl_mismatching_register_type_and_variable_type | "error: unsupported resource register binding '%select{t\|u\|b\|s}0' on variable of type '%1'"| Emitted if a variable type that isn't accompanied with a resource class attribute is bound with a standard register type (`t`, `u`, `b`, or `s`). |
 | err_hlsl_unsupported_register_type_and_variable_type | "error: register binding type '%0' not supported for variable of type '%1'" | Emitted if a variable type is bound using an unsupported register type, like binding a float with the 'x' register type. |
 | err_hlsl_mismatching_register_type_and_resource_type | "error: %select{srv\|uav\|cbv\|sampler}2 type '%0' requires register type '%select{t\|u\|b\|s}2', but register type '%1' was used." | Emitted if a known resource type is bound using a standard but mismatching register type, e.g., RWBuffer<int> getting bound with 's'.|
-| err_hlsl_unsupported_register_type_and_resource_type | "error: invalid register type '%0' used; expected 't', 'u', 'b', or 's'"| Emitted if an unsupported register type is used to bind a resource, like 'y' being used with RWBuffer<int>.|
+| err_hlsl_unsupported_register_type_and_resource_type | "error: invalid register type '%0' used; expected 't', 'u', 'b', or 's'"| Emitted if an unsupported register type is used to bind a UDT.|

```

Comment:
"err_hlsl_unsupported_register_type_and_resource_type" - do these diagnostic names appear anywhere user-visible?  I don't think the name really matches what it is now.  I'd expect "user defined type" to be in the name somewhere, for example.  Not sure "resource type" belongs at all.

Reply:
They do not, but because UDTs are now the only use, it makes sense to update the name.